### DISCUSSION TITLE
How can I bundle a GraphQL schema?

### DISCUSSION THREAD

**Discussion by oliversalzburg** [parcel-bundler/parcel#6873] [Q&A] [answered]
I'm trying to bundle a GraphQL schema as a string, to provide it to a GraphiQL component.

I simply tried:
```ts
import schemaSdl from "./schema.docs.graphql";
```
> I will have to manually install `@parcel/transformer-graphql` due to autoinstall not working with yarn2 and the `latest` tag is currently off for this package. `2.0.0-rc.0` needs to be installed explicitly.

`parcel` will then complain:
```shell
@parcel/transformer-graphql: Only executable definitions are supported in GraphQL Documents.
```

Why does it matter if the definition is executable? The [documentation on importing GraphQL](https://v2.parceljs.org/getting-started/migration/#importing-graphql) says that the document is returned as a string.

So I try to load the data via `fs`:
```ts
const schemaSdl = fs.readFileSync(path.resolve(__dirname, "./docs.graphql.schema"), "utf8");
```

The [documentation on fs.readFileSync calls](https://v2.parceljs.org/features/node-emulation/#%F0%9F%93%84-inlining-fs.readfilesync) says that these will be inlined as strings for browser targets. But when I load the bundled application, I get a `readFileSync is not a function` error.

What is the right method to do this?

**Comment by mischnic** [comment, marked as answer]
> const schemaSdl = fs.readFileSync(path.resolve(__dirname, "./docs.graphql.schema"), "utf8");

The docs don't say this explicitly, but you need to use `fs.readFileSync(path.join(__dirname, "something"), "utf8)` instead.

**Reply by oliversalzburg** [reply]
That works perfectly. Thanks a lot.

Can you explain to me why `path.join` is required here instead of `path.resolve`?

**Reply by mischnic** [reply]
That's simply what Parcel detects statically (and only that, for simplicity)

**Comment by ryangoree** [comment]
Same question regarding the error:

```
@parcel/transformer-graphql: Only executable definitions are supported in GraphQL Documents.
```

why does it matter if the definition is executable?

I want to be able to export typeDefs for use by other packages/apps.

**Comment by vbutani** [comment]
I am also facing the same error. I am trying to migrate my project from webpack to Parcel 2, but stuck with this error!
`@parcel/transformer-graphql: Only executable definitions are supported in GraphQL Documents.`
